### ISSUE TITLE
Support Scala 2.13-RC1?

### ISSUE THREAD

**Issue by mslinn** [pureconfig/pureconfig#467]
Would be helpful :)

**Comment by ruippeixotog** [comment]
I have this on my plans :) I already started, but I hit some issues configuring SBT for this. I'll try again when I have time.

**Comment by mslinn** [comment]
I also hit issues. Here is a bit of code I added to `build.sbt` for my [Introduction to Scala](https://www.scalacourses.com/showCourse/40) course that might be helpful:

```
libraryDependencies ++=
  scalaVersion {
    case sv if sv.startsWith("2.13") => List(
      "org.scala-lang.modules" %% "scala-parallel-collections" % "0.2.0"
    )

    case _ => Nil
  }.value
updateOptions := updateOptions.value.withCachedResolution(true)
```

BTW, the course has a lecture on `PureConfig`.

**Comment by leifwickland** [comment]
@ruippeixotog I also started poking at building for 2.13, I but haven't made much progress. I suspect you're further than I am. I you push your branch, perhaps I could help as I have time.

**Comment by leifwickland** [comment]
@ruippeixotog @mslinn, My sticking point was that I couldn't find a `scalacheck-shapeless` that's compatible with `2.13.0-RC1`. Did you guys get past that one?

**Comment by mslinn** [comment]
@leifwickland Perhaps @alexarchambault can help with a version of `scalacheck-shapeless` for Scala 2.13-RC1?

**Comment by alexarchambault** [comment]
@mslinn It's on its way: https://github.com/alexarchambault/scalacheck-shapeless/pull/103.

**Comment by alexarchambault** [comment]
Done, `1.2.2` is out: http://repo1.maven.org/maven2/com/github/alexarchambault/scalacheck-shapeless_1.14_2.13.0-RC1/1.2.2/.

**Comment by mslinn** [comment]
@alexarchambault Wow! Thanks!

**Comment by leifwickland** [comment]
Thanks, @alexarchambault. That got me a little further.

**Comment by leifwickland** [comment]
@ruippeixotog, I pushed my very incomplete work of getting to 2.13.0-RC1. https://github.com/pureconfig/pureconfig/tree/hacking-toward-2.13.0-RC1

The immediate blocker is that `CanBuildFrom` is removed from 2.13. I think the party line is that [`scala-collection-compat`](https://github.com/scala/scala-collection-compat) is supposed to be the answer, but currently there's no blessed release version. 1.0.0 has a bug and they say they'll have to release 2.0.0 to fix it.

There is supposedly a `scalafix` that will rewrite source to use it and get compat. 

Trying that out is what I'd do next.

**Comment by ruippeixotog** [comment]
Thanks, Leif! I already had everything compiling successfully on Scala 2.13-RC1 (with my own compat layer, which can easily be replaced if an official one appears). My troubles were with cross-compiling the project on SBT as I was getting some weird dependency errors. I’ll try again this week looking at your build, thanks for sharing!

**Comment by leifwickland** [comment]
@ruippeixotog Ah, in that case you'll get no benefit from mine. All I could possibly have is newer versions of some deps that have been published very recently. If I can be of assistance let me know.

**Comment by skestle** [comment]
Can we really consider this closed when it's not published to a repository and https://github.com/scala/make-release-notes/blob/2.13.x/projects-2.13.md is not updated?

(I'm expecting to see it at https://oss.sonatype.org/content/repositories/snapshots/com/github/pureconfig/)

**Comment by ruippeixotog** [comment]
You’re right that we should release PureConfig in order to rightly close this. I’ll take care of that later today 👍

We can proceed later to do a PR to make-release-notes adding PureConfig, but I wouldn’t consider it a requirement.

**Comment by mslinn** [comment]
"The job is not done until the paperwork is finished". Any release notes that could be written would be helpful to me, so I can update the ScalaCourses lecture on PureConfig without losing too much hair.

**Comment by ruippeixotog** [comment]
@skestle I just released PureConfig v0.11.0, it should be available on Maven in one hour or two.

@mslinn our release notes are written in [CHANGELOG.md](https://github.com/pureconfig/pureconfig/blob/master/CHANGELOG.md).
